internal/pkg/service/stream/api/service/task.go

```diff
@@ -43,13 +45,19 @@ func (s *service) startTask(cfg taskConfig) (task.Task, error) {
 	}
 
 	taskID := task.ID(cfg.Type + "/" + objectKey)
-	return s.tasks.RunTask(task.Config{
+	return s.tasks.StartTask(ctx, task.Config{
 		Type: cfg.Type,
 		Key:  task.Key{ProjectID: cfg.ProjectID, TaskID: taskID},
 		Lock: cfg.ObjectKey.String(),
 		Context: func() (context.Context, context.CancelFunc) {
-			return context.WithTimeout(context.Background(), cfg.Timeout)
+			return context.WithTimeout(context.WithoutCancel(ctx), cfg.Timeout)
+		},
```

Comment:
Here is factory of the task background context, we can use relatively new `context.WithoutCancel` call, I will fix it also in the other places.